BUILD_TOPOLOGY.toml

```diff
@@ -0,0 +1,328 @@
+type = "per-arch"  # Built per GPU architecture
+artifact_group_deps = ["hip-runtime", "profiler-core"]  # Many libs use rocprofiler-sdk for annotations
+
+[artifact_groups.ml-libs]
+description = "Machine learning libraries"
+type = "per-arch"
+artifact_group_deps = ["math-libs", "profiler-core"]
+
+[artifact_groups.comm-libs]
+description = "Communication libraries"
+type = "per-arch"
+artifact_group_deps = ["hip-runtime"]
+
+[artifact_groups.profiler-core]
+description = "Core profiling libraries and annotation support"
+type = "generic"
+artifact_group_deps = ["core-runtime"]
+
+[artifact_groups.dctools-core]
+description = "Data center management tools with minimal dependencies"
+type = "generic"
+artifact_group_deps = ["core-runtime", "profiler-core"]
+
+# Future artifact groups
+# [artifact_groups.dctools-rocm]
+# description = "Data center tools that depend on ROCm libraries"
+# type = "generic"
+# artifact_group_deps = ["math-libs", "ml-libs", "profiler-core"]
+
+[artifact_groups.profiler-apps]
+description = "Profiler applications and analysis tools"
+type = "generic"
+artifact_group_deps = ["profiler-core", "compiler"]
+
+# [artifact_groups.iree-libs]
+# description = "IREE and Fusilli integration"
+# type = "generic"
+# artifact_group_deps = ["third-party-sysdeps", "base"]
+
+# ==============================================================================
+# ARTIFACTS - The actual build outputs that can be transmitted between stages
+# ==============================================================================
+# Artifacts are the fundamental packaging units
+# Each artifact produces components: dbg, dev, doc, lib, run, test
+
+# --- Third-party Dependencies ---
+
+[artifacts.sysdeps]
+artifact_group = "third-party-sysdeps"
+type = "target-neutral"
+artifact_deps = []  # No dependencies on other artifacts
+feature_group = "CORE"  # Part of core, enabled by default
+
+[artifacts.host-blas]
+artifact_group = "third-party-libs"
+type = "target-neutral"
+artifact_deps = []
+feature_group = "HOST_MATH"  # Controlled by THEROCK_ENABLE_HOST_MATH (OFF by default)
+
+[artifacts.host-suite-sparse]
+artifact_group = "third-party-libs"
+type = "target-neutral"
+artifact_deps = ["host-blas"]  # SuiteSparse depends on BLAS
+feature_name = "HOST_SUITE_SPARSE"
+feature_group = "HOST_MATH"  # Controlled by THEROCK_ENABLE_HOST_MATH (OFF by default)
+disable_platforms = ["windows"]
+
+[artifacts.fftw3]
+artifact_group = "third-party-libs"
+type = "target-neutral"
+artifact_deps = []
+feature_group = "HOST_MATH"  # Controlled by THEROCK_ENABLE_HOST_MATH (OFF by default)
+
+# --- Base Infrastructure ---
+
+[artifacts.base]
+artifact_group = "base"
+type = "target-neutral"
+artifact_deps = []
+feature_group = "CORE"  # Part of core, enabled by default
+
+# --- Compiler Artifacts ---
+
+[artifacts.amd-llvm]
+artifact_group = "compiler"
+type = "target-neutral"
+artifact_deps = ["sysdeps"]
+feature_name = "COMPILER"
+feature_group = "ALL"
```

Comment:
Are these fields case sensitive? Will they always be all lowercase or all uppercase? Should the scripts do some validation on casing?